feat: Show event distance on search cards

dave-ok opened the pull request:
![image](https://user-images.githubusercontent.com/19638455/190673696-bcde0464-656d-4d6c-90a6-4543115ebc59.png)

silksil commented on src/hooks/useAccommodationsAndOffers.tsx/index.ts:
```diff
@@ -36,16 +54,70 @@ export const useAccommodationsAndOffers = (props: SearchTypeProps | void) => {
+  // determine from search url if there is a relevant area to focus
+  let focusedEventCoordinates;
+
+  // if search url contains a focusedEvent get the coordinates
```
discussed with dave that it is more logical to move this logic to `useCurrentEvents`

silksil commented on src/components/SearchCard.tsx:
```diff
@@ -8,16 +8,18 @@ import { useWindowsDimension } from '../hooks/useWindowsDimension';
 }
-const pricePerNight = (prices: number[], numberOfDays: number) =>
-  (Math.min(...prices) / numberOfDays).toFixed(2);
+/* const pricePerNight = (prices: number[], numberOfDays: number) =>
```
it is correct that this is commented out?

dave-ok commented:
@silksil I have moved every logic that couples events with accommodations out of the `useAccomodationsAndOffers`.
I instead added an optional callback parameter to the hook's parameter, to enable transforming the `accommodation` object.
The function must still return a type of `AccommodationWithId` object, so nothing is broken.

I also dropped the `useCurrentEvents` hook and changed it to a regular function `getCurrentEventsWithinRadius`. It didn't seem to be necessary as a hook. Any component using this function could memoize it if needed.

silksil commented on src/components/MapBox.tsx:
```diff
@@ -119,61 +134,46 @@ export const MapBox: React.FC = () => {
   const { accommodations, coordinates, isLoading, latestQueryParams, isFetching, error } =
-    useAccommodationsAndOffers();
+    useAccommodationsAndOffers(
+      undefined,
```
instead of passing a second prop, it is maybe a bit cleaner to pass it as part of the `SearchProps` object. 

Then you don't have to pass an `undefined` here.

dave-ok replied:
Besides breaking the logic for `useQuery`, adding this to `SearchProps` will require making all other fields of `SearchProps` optional for cases where no other fields are available. Made the parameter into an object instead with a separate field for `accommodationtransformFn`

silksil commented on src/components/MapBox.tsx:
```diff
@@ -119,61 +134,46 @@ export const MapBox: React.FC = () => {
+    [searchParams]
+  );
 
   // TODO: replace this with activeAccommodations
```
this is a comment that i included a while back, but can removed now.

silksil commented on src/components/Results.tsx:
```diff
@@ -35,7 +37,18 @@ const StyledContainer = styled(Box)(({ theme }) => ({
+  );
+
+  // apply a callback function to transform returned accommodation objects
+  const { accommodations, isFetching, latestQueryParams } = useAccommodationsAndOffers(
```
You also pas the `focusedEvent` in the `MapBox` component. 

Why is this done twice?

dave-ok replied:
`Results` and `MapBox` are 2 entirely independent components. They can't reuse this value between them

silksil commented on src/hooks/useAccommodationsAndOffers.tsx/index.ts:
```diff
@@ -36,16 +58,48 @@ export const useAccommodationsAndOffers = (props: SearchTypeProps | void) => {
-    () => allAccommodations.filter((a) => a.offers.length > 0),
-    [allAccommodations]
-  );
+  const accommodations = useMemo(() => {
```
You could consider including an comment above `accommodation` stating what is happening here.

Also, I think we should eventually consider creating a helper function that those these transformations, so we keep the `index.ts` clean.  Something like

`accommodations = getNormalizedAccommodations(accommodations, latesQueryParams)`

silksil commented on src/utils/events.ts:
```diff
@@ -58,3 +60,41 @@ export const getEventsWithinRadius = (
+}
+
+// get active events within a radius with optional focused event
+export const getCurrentEventsWithinRadius = ({
```
maybe `getActiveEventsWithinRadius` is a bit more descriptive.
